tests/Api/Category/UpsertCategoryIntegration.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Akeneo\Pim\tests\Api\Category;
+
+use Akeneo\Pim\tests\Api\ApiTestCase;
+
+class UpsertCategoryIntegration extends ApiTestCase
+{
+    public function testUpsertDoingUpdate()
+    {
+        $api = $this->createClient()->getCategoryApi();
+
+        $response = $api->upsert('sandals', [
+            'parent' => 'winter_collection',
+            'labels' => [
+                'en_US' => 'Sandals !',
+                'fr_FR' => 'Sandales',
+            ],
+        ]);
+
+        $this->assertSame(204, $response);
+
+        $category = $api->get('sandals');
+        $this->assertSameContent([
+            'code' => 'sandals',
+            'parent' => 'winter_collection',
+            'labels' => [
+                'en_US' => 'Sandals !',
+                'fr_FR' => 'Sandales',
+            ],
+        ], $category);
+    }
+
+    public function testUpsertDoingCreate()
+    {
+        $api = $this->createClient()->getCategoryApi();
+        $response = $api->upsert('booties', [
+            'parent' => 'summer_collection',
+            'labels' => [
+                'en_US' => 'Booties',
+                'fr_FR' => 'Bottines',
+            ],
+        ]);
+
+        $this->assertSame(201, $response);
+
+        $category = $api->get('booties');
+        $this->assertSameContent([
+            'code' => 'booties',
+            'parent' => 'summer_collection',
+            'labels' => [
+                'en_US' => 'Booties',
+                'fr_FR' => 'Bottines',
+            ],
+        ], $category);
+    }
```

Comment:
test failing upsert as well: problem on validation in the updater (wrong data type for example) and a problem in the Symfony validation.
This way, we have 1 test on each main exception.